### ISSUE TITLE
Why number of predicted classes is different from DPN Paper?

### ISSUE THREAD

**Issue by Chandan-IITI** [Lackel/LOOP#2]
Hi @Lackel ,

It ia good work. I have a query as I am unable to understand through code and paper that 

**if the pretraining stage is idnetical in the case of your earlier work DPN and LOOP then why model will predict different number of classes for the datasets?** 

What I can see in the paper  and code that pretraining stage is identical abd prediction of number of classes are performed before the main training loop then why it is differently predicting. 

**Am I missing something?**

**Comment by Lackel** [comment]
Hi, thanks for your interest in our work.

Actually, the pretraining stage of DPN and LOOP has a slight difference, the outer pretraining loop of DPN is MLM loss on the semi-dataset (for step, batch in enumerate(self.data.pretrain_semi_dataloader)), while the outer pretraining loop of LOOP is CE loss on labeled data (for step, batch in enumerate(tqdm(data.train_labeled_dataloader, desc="Iteration"))). This difference leads to a different prediction of the number of categories.
